great_expectations/core/run_identifier.py

```diff
@@ -54,9 +54,7 @@ def __init__(
         if not run_time.tzinfo:
             # this takes the given time and just adds timezone (no conversion)
             run_time = run_time.replace(tzinfo=datetime.timezone.utc)
```

Comment:
Wait, this is the bug, isn't it? if run_time is naive and you replace the timezone with utc, you fundamentally change the time if the naive time isn't utc. we should be making this case impossible by only making (preferably utc) aware times or calling `run_time = run_time.astimezone(tz=timezone.utc)`

Reply:
Does that mean we should just remove this do you think? So if somehow it was naive it stays naive?

Reply:
Alternatively, we could assume that if it's naive it is local I guess...

Reply:
this definitely has to go because it's destructive--I also think we should stick to utc aware times, especially if the string output doesn't contain full tz information (looks like we're just including offsets)--that will lead to confusing behavior when offsets are reinterpreted by browsers or anything with more specific timezone info

Reply:
I'm not sure what you mean by being destructive (or the difference of full tz info vs. offsets)...can you clarify the concern a bit more?

Reply:
Agreed...curious what Drew thinks here. End of the day as long as it's an aware datetime, the underlying problem is fixed from my perspective, but happy to only do the localization at the end.